command/meta.go

```diff
@@ -427,3 +428,173 @@ type funcVar func(s string) error
+	},
+	"job status single": {
+		Path:        "/jobs/:jobID@:namespace",
+		Description: "View job details and metrics",
+	},
+	"job run": {
+		Path:        "/jobs/:jobID@:namespace",
+		Description: "View this job",
+	},
+	"alloc status": {
+		Path:        "/allocations/:allocID",
+		Description: "View allocation details",
+	},
+	"var list": {
+		Path:        "/variables",
+		Description: "View Nomad variables",
+	},
+	"var list prefix": {
+		Path:        "/variables/path/:prefix",
+		Description: "View Nomad variables at this path",
+	},
+	"var get": {
+		Path:        "/variables/var/:path@:namespace",
+		Description: "View variable details",
+	},
+	"var put": {
+		Path:        "/variables/var/:path@:namespace",
+		Description: "View variable details",
+	},
+	"job dispatch": {
+		Path:        "/jobs/:dispatchID@:namespace",
+		Description: "View this job",
+	},
+	"eval list": {
+		Path:        "/evaluations",
+		Description: "View evaluations",
+	},
+	"eval status": {
+		Path:        "/evaluations?currentEval=:evalID",
+		Description: "View evaluation details",
+	},
+	"deployment status": {
+		Path:        "/jobs/:jobID/deployments",
+		Description: "View all deployments for this job",
+	},
+}
+
+func (m *Meta) formatUIHint(url string, description string) string {
+	if description == "" {
+		description = defaultHint
+	}
+
+	description = fmt.Sprintf("%s in the Web UI:", description)
+
+	// Basic version without colors
+	hint := fmt.Sprintf("%s%s %s", uiHintDelimiter, description, url)
+
+	// If colors are disabled, return basic version
+	_, coloredUi := m.Ui.(*cli.ColoredUi)
+	if m.noColor || !coloredUi {
+		return hint
+	}
+
+	return fmt.Sprintf("%[1]s%[2]s%[3]s%[4]s%[5]s %[6]s%[7]s%[8]s",
+		bold,
+		magenta,
+		uiHintDelimiter[1:], // "==> "
+		description,
+		resetter,
+		blue,
+		url,
+		resetter,
+	)
+}
+
+func (m *Meta) buildUIPath(route UIRoute, params map[string]string) (string, error) {
+	client, err := m.Client()
+	if err != nil {
+		return "", fmt.Errorf("error getting client config: %v", err)
+	}
+
+	path := route.Path
+	for k, v := range params {
+		path = strings.ReplaceAll(path, fmt.Sprintf(":%s", k), v)
+	}
+
+	return fmt.Sprintf("%s/ui%s", client.Address(), path), nil
+}
+
+func (m *Meta) showUIPath(ctx UIHintContext) (string, error) {
+	route, exists := CommandUIRoutes[ctx.Command]
+	if !exists {
+		return "", nil
+	}
+
+	url, err := m.buildUIPath(route, ctx.PathParams)
+	if err != nil {
+		return "", err
+	}
+
+	if ctx.OpenURL {
+		if err := util.OpenURL(url); err != nil {
+			m.Ui.Warn(fmt.Sprintf("Failed to open browser: %v", err))
+		}
+	}
+
+	if m.uiHintsDisabled() {
+		return "", nil
+	}
+
+	return m.formatUIHint(url, route.Description), nil
+}
+
+func (m *Meta) uiHintsDisabled() bool {
+	client, err := m.Client()
+	if err != nil {
+		return true
+	}
+	agent, err := client.Agent().Self()
+	if err != nil {
+		return true
+	}
+	agentConfig := agent.Config
+	uiConfig, ok := agentConfig["UI"].(map[string]interface{})
```

Comment:
I think this needs to be locally configurable instead of clusterwide configurable. For example it's probably desirable on operator machines and bastion hosts, but potentially undesirable on actual Nomad Clients where the CLI is only used for maintenance/debugging.

Since the UI is often proxied across many agents, if not the entire cluster, configuring this parameter clusterwide seems difficult.

I think we probably want to implement something like #24706 or at least an env var to let this behavior be user configurable.

We should probably be careful about doing implicit API calls from the CLI as well. Even though `Agent().Self()` should be cheap for the responding Agent, users running scripts over potentially slow links may see a significant slowdown from the extra roundtrips.